src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -520,7 +520,10 @@ private void OnAsynchronousRequestCancelled(object state)
             }
         }
 
-        private sealed class Request
+        /// <remarks>
+        /// This inherits from <see cref="TaskCompletionSource{TResult}"/> to avoid allocating two objects when we can just use one.
+        /// </remarks>
```

Comment:
Ah, i know why i dont' like this.  Because now people holding onto a 'Request' might accidently call TCS methods that might subvert our own logic.  Containment means we really control exactly the API that we expect consumers of Request to use.  

Let's chat if you feel like this isn't an issue.